be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala

```diff
@@ -20,10 +20,13 @@ trait MessageHandler extends AskPatternConstants {
     */
   def dbActor: AskableActorRef = DbActor.getInstance
 
-  def checkParameters(rpcRequest: JsonRpcRequest, errorMsg: String): Future[GraphMessage] = {
+  def checkParameters[T](rpcRequest: JsonRpcRequest, errorMsg: String): Future[(GraphMessage, Message, Option[T])] = {
```

Comment:
As I said on slack, for me the function looks good. 
I suggest changing the name, since now the function does not only check, but also extract the message and data. Something like `extractParameters` ?

Reply:
sounds good to me

Reply:
Also one more comment, we can do the same for the function `checkLao` in term of extraction of channel

Reply:
addressed thanks!